docs/proposal/kernel_native_mode_restart.md

```diff
@@ -0,0 +1,119 @@
+
+## Motivation
+
+Using kmesh for mesh acceleration in a K8s cluster, ensuring that acceleration capabilities remain uninterrupted during kmesh restart scenarios provides a significant competitive advantage through seamless restarts and upgrades.
+
+### Objectives
+
+1. Persistently save related configurations locally during kmesh restart and automatically manage them to maintain uninterrupted acceleration capabilities.
+2. Automatically restore related configurations and update them after a restart.
+
+### Proposal
+
+Implement configuration persistence management and service continuity.
+
+#### Configuration Persistence Management
+
+- Determine if kmesh is shutting down normally or  restart. If it's a restart, persist relevant configurations to a specified directory.
+- Persist the eBPF programs used for service traffic management, allowing independent traffic governance even after kmesh shuts down, thus ensuring service continuity.
+- Persist other relevant functional configurations:
+  - Management features: Automatically pull the latest configurations and refresh them after each restart.
+  - Certificate subscription: Reacquire certificates after each restart.
+
+#### Configuration Restoration and Update
+
+- On kmesh startup, determine if it is a fresh start or a restart. If it’s a restart, restore configurations from the specified directory and compare them with the latest received configurations for updates.
+
+### Limitations
+
+Currently, upgrade scenarios are not supported but will be in the future.
+
+If kmesh experiences a coredump leading to a restart, configuration persistence capabilities cannot be achieved.
+
+## Design Details
+
+### Configuration Persistence Management
+
+<div align="center">
+
+![kernel_native_mode_restart](pics/kernel_native_mode_restart.svg)
+
+</div>
+
+- `ebpf_prog` is used for traffic governance operations after kmesh shuts down.
+- `ebpf_map` records configurations for traffic governance operations after kmesh shuts down.
+- `inner_map_mng` tracks information during the serialization process to manage `inner_map`, aiding in correctly handling the relationships between `ebpf_map` during kmesh restart recovery.
+- `hashName` records the hash value of each XDS configuration tree for comparison to detect changes.
+- `kmesh_version` records the version information of kmesh for comparing whether kmesh is in a restart or upgrade state.
+- `tail_call_map` tracks eBPF tail call information, recording tail call programs.
+
+#### Persistence Operations
+
+- `ebpf_prog` with `sockconn`, `sockops`, and `tracepoint` ensures that eBPF traffic governance capabilities remain uninterrupted after being solidified.
+  
+  - Use `bpf_link` to pin the attached `sockconn`/`sockops` programs.
+  
+  - Directly pin the tracepoint programs to the specified directory.
+
+- Pin other eBPF maps directly to the specified directory.

```

Comment:
ditto. 
And it can be shown that additional operations are required in the `kind cluster` to achieve this result.